src/close_fds.c

```diff
@@ -22,6 +22,12 @@
 
 #include <sys/stat.h>
 
+#ifdef __FreeBSD__
+#define OPEN_FILES_DIR "/dev/fd"
+#else
+#define OPEN_FILES_DIR "/proc/self/fd"
```

Comment:
Sorry for being late to the party, but `/dev/fd` usually works on Linux, too, since it's a symlink to `/proc/self/fd` (at least that's the case with Fedora 38 on my laptop.

Update: this is done that way since systemd v12, here's the commit from 2010: https://github.com/systemd/systemd/commit/5c0532d1cc989d2f78d2cd4a18058daa58143705.